ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js

```diff
@@ -0,0 +1,73 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import PageContainerFooter from '../../../../components/ui/page-container/page-container-footer'
+import TextField from '../../../../components/ui/text-field'
+import { CONTACT_LIST_ROUTE } from '../../../../helpers/constants/routes'
+import { isValidAddress } from '../../../../helpers/utils/util'
+
+export default class AddContact extends PureComponent {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    addressBook: PropTypes.object,
+    addToAddressBook: PropTypes.func,
+    history: PropTypes.object,
+  }
+
+  state = {
+    nickname: '',
+    address: '',
+  }
+
+   render () {
+    const { nickname, address } = this.state
+    const { history, addToAddressBook } = this.props
+
+    return (
+      <div className="settings-page__content-row">
+        <div className="settings-page__content-item">
+          <div className="settings-page__content-item-col">
+
+            <TextField
+              type="text"
+              id="nickname"
+              placeholder={this.context.t('addAlias')}
+              value={nickname}
+              onChange={e => this.setState({ nickname: e.target.value })}
+              fullWidth
+              margin="dense"
+            />
+
+            <TextField
+              type="text"
+              id="address"
+              placeholder={this.context.t('addEthAddress')}
+              value={address}
+              onChange={e => this.setState({ address: e.target.value })}
+              fullWidth
+              margin="dense"
+            />
+            </div>
+          </div>
+        <PageContainerFooter
+          cancelText={this.context.t('cancel')}
+          onSubmit={() => {
+            if (isValidAddress(address)) {
+              addToAddressBook(address, nickname)
+              history.push(CONTACT_LIST_ROUTE)
+            } else {
+              this.setState({ address: 'invalid address' })
```

Comment:
Can we put the `'invalid address'` message in an error field. `TextField` takes an `error` prop.

We could store an `error` property in state, and set that here, and clear that property when the address is valid. Then we could pass that state property to the error prop of the relevant `TextField`